index.sh

```diff
@@ -226,35 +227,45 @@ elif command -v update-alternatives >/dev/null 2>&1 ; then
   altcmd="update-alternatives"
 fi
 
+doRegister="n"
 if [ "${altcmd:-x}" != "x" ] ; then
   wrkdir="$(pwd)"
   
-  isatty=0
-  if [ -t 0 ] || [ -t 2 ] ; then # When piping into get micro, e.g. `curl https://getmic.ro | `
-    isatty=1
-  fi
-  if [ "${GETMICRO_REGISTER:-x}" = "n" ] || [ "${GETMICRO_REGISTER:-x}" = "N" ] ; then
+  if echo "x${GETMICRO_REGISTER:-x}" | grep -Eqie '^xn(o)?$' 1>/dev/null 2>&1 ; then
     doRegister="n"
-  elif [ "${GETMICRO_REGISTER:-x}" = "y" ] || [ "${GETMICRO_REGISTER:-x}" = "Y" ] ; then
+  elif echo "x${GETMICRO_REGISTER:-x}" | grep -Eqie '^xy(es)?$' 1>/dev/null 2>&1 ; then
     doRegister="y"
-  elif [ $isatty -eq 1 ] ; then
+  elif [ -t 0 ] || [ -t 2 ] ; then # Check if there is a user viewing this message
     cat 1>&2 << 'EOM'
 /=====================================\\
-|     update-alternatives detected     |
+|   update-alternatives is supported   |
 \\=====================================/
 
+TL;DR: This is an optional feature that is safely disabled by default. If this
+ is confusing or you are short on time, you can skip reading this message.
+ 
+(Notice: if you y use this feature and you put micro in a location unaccessible
+ to other users, this will break many tools such as crontab for other users.
+ Please run `cd /usr/bin` prior to installation to prevent this conundrum.)
+
 getmicro can use update-alternatives to register micro as a system text editor.
 For example, this will allow `crontab -e` open the cron file with micro.
 
-To avoid this prompt in the future, define the GETMICRO_REGISTER variable. E.x:
+To enable this opt-in feature, define the GETMICRO_REGISTER variable. E.x:
 
-  $ curl https://getmic.ro | GETMICRO_REGISTER=y sh
+  $ su - root -c "cd /usr/bin; wget -O- https://getmic.ro | GETMICRO_REGISTER=y sh"
   
-Many people find it useful to make micro available on the PATH. One way to do
- this is to enter a root shell and run `cd /usr/bin` prior to installation.
+Alternatively, https://getmic.ro/r will define GETMICRO_REGISTER=y for you:
+
+  $ cd /usr/bin
+  $ curl https://getmic.ro/r | sudo sh
 
 EOM
-    cpt="Register '$wrkdir/micro' with update-alternatives [y/n]: "
+    doRegister="n"
+    cat >/dev/null << 'NUL'
+    # Commenting this out resolves https://github.com/benweissmann/getmic.ro/pull/32#discussion_r800962643
+    
+    cpt="Register '$wrkdir/micro' with update-alternatives (prefer n if unsure) [y/N]: "
```

Comment:
I though the idea here was that we wouldn't prompt unless the user is root / can run update-alternatives? If i'm reading this correctly, we still end up prompting non-root users here, no?

Reply:
All of this is in a comment block starting on line 265. No user gets prompted for input ever.

All users, regardless of whether they are root or not, will receive the info blurb about update-alternatives if its available on their system.